src/java/org/apache/cassandra/service/StorageService.java

```diff
@@ -4209,7 +4209,10 @@ private LinkedHashMap<InetAddressAndPort, Float> getEffectiveOwnership(String ke
                 for (String keyspaceName : userKeyspaces)
                 {
                     if (!Schema.instance.getKeyspaceInstance(keyspaceName).getReplicationStrategy().hasSameSettings(replicationStrategy))
-                        throw new IllegalStateException("Non-system keyspaces don't have the same replication settings, effective ownership information is meaningless");
+                        throw new IllegalStateException(String.format(
+                        "Non-system keyspaces: %s and %s don't have the same replication settings, effective ownership information is meaningless",
```

Comment:
@maoling you might probably gather all keyspaces for which replication settings are different and enrich exception message with them?

However, there might be in theory dozens of keyspaces and it might pollute the output unnecessarily. Not sure how to control that.